tests/fixtures/extended-tests.json

```diff
@@ -46,10 +46,16 @@
             ["/base{/group_id,first_name}/pages{/page,lang}{?format,q}","/base/12345/John/pages/5/en?format=json&q=URI%20Templates"],
             ["/sparql{?query}", "/sparql?query=PREFIX%20dc%3A%20%3Chttp%3A%2F%2Fpurl.org%2Fdc%2Felements%2F1.1%2F%3E%20SELECT%20%3Fbook%20%3Fwho%20WHERE%20%7B%20%3Fbook%20dc%3Acreator%20%3Fwho%20%7D"],
             ["/go{?uri}", "/go?uri=http%3A%2F%2Fexample.org%2F%3Furi%3Dhttp%253A%252F%252Fexample.org%252F"],
-            ["/service{?word}", "/service?word=dr%C3%BCcken"],
-            ["/lookup{?Stra%C3%9Fe}", "/lookup?Stra%C3%9Fe=Gr%C3%BCner%20Weg"],
-            ["{random}" , "%C5%A1%C3%B6%C3%A4%C5%B8%C5%93%C3%B1%C3%AA%E2%82%AC%C2%A3%C2%A5%E2%80%A1%C3%91%C3%92%C3%93%C3%94%C3%95%C3%96%C3%97%C3%98%C3%99%C3%9A%C3%A0%C3%A1%C3%A2%C3%A3%C3%A4%C3%A5%C3%A6%C3%A7%C3%BF"],
-            ["{?assoc_special_chars*}", "?%C5%A1%C3%B6%C3%A4%C5%B8%C5%93%C3%B1%C3%AA%E2%82%AC%C2%A3%C2%A5%E2%80%A1%C3%91%C3%92%C3%93%C3%94%C3%95=%C3%96%C3%97%C3%98%C3%99%C3%9A%C3%A0%C3%A1%C3%A2%C3%A3%C3%A4%C3%A5%C3%A6%C3%A7%C3%BF"]
+            ["/service{?word}", [ "/service?word=dr%C3%BCcken", "/service?word=dr%C3%83%C2%BCcken"]],
+            ["/lookup{?Stra%C3%9Fe}", ["/lookup?Stra%C3%9Fe=Gr%C3%BCner%20Weg", "/lookup?Stra%C3%9Fe=Gr%C3%83%C2%BCner%20Weg"]],
+            ["{random}",
+                [ "%C5%A1%C3%B6%C3%A4%C5%B8%C5%93%C3%B1%C3%AA%E2%82%AC%C2%A3%C2%A5%E2%80%A1%C3%91%C3%92%C3%93%C3%94%C3%95%C3%96%C3%97%C3%98%C3%99%C3%9A%C3%A0%C3%A1%C3%A2%C3%A3%C3%A4%C3%A5%C3%A6%C3%A7%C3%BF",
```

Comment:
Did you modify these by hand or copy them from the upstream source?

Reply:
By hand.

Reply:
So these are just copies of https://github.com/uri-templates/uritemplate-test iirc and I'd rather not bee adding data to it manually. Would it make more sense to force the same behaviour across versions rather than introducing behaviour that's otherwise undocumented to users as a result of using the stdlib?

Reply:
I can do that, but how do you force it?

Reply:
For older versions of python, we'd want `~` to be considered safe if I'm understanding the documentation properly. So we'd pass `safe='/~'` to replicate the existing behaviour since that is documented as _extending_ the default list of points that are not quoted.

Reply:
Is that the issue, though? Looking at the `"service{?word}"` example, the change is in the encoding of `drücken` which has nothing to do with `~` from what I can tell (whatever `%83%C2` represents).

Reply:
I haven't the foggiest clue what's happening:

```
 .tox/py38/bin/python 
Python 3.8.0 (default, Dec  5 2019, 17:23:04) 
[GCC 9.2.1 20190827 (Red Hat 9.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import rfc3986
>>> rfc3986.urlparse("drücken")
ParseResult(scheme=None, userinfo=None, host=None, port=None, path='dr%C3%BCcken', query=None, fragment=None)
>>> import urllib
>>> urllib.parse.quote("drücken")
'dr%C3%BCcken'
```

It's unclear what's happening in this test

Reply:
Yay, mysteries. 😜

I'll see if I can track down what's causing this in the code.

Reply:
Turns out the issue was because the JSON test file was being open without a specified encoding which was causing mojibake under Windows.